packages/svelte/tests/runtime-runes/samples/await-resolve/main.svelte

```diff
@@ -0,0 +1,18 @@
+<script>
+	import Pending from "./Pending.svelte"
+	import Then from './Then.svelte';
+
+	const promise_a = Promise.resolve('a')
+	const promise_b = Promise.resolve('b')
+
+	let current_promise = $state(promise_a);
+</script>
+
+{#await current_promise}
+	<Pending />
+{:then value}
+	<Then {value} />
```

Comment:
This is what I did to ensure that it doesn't even render - also simplifies the test because you don't need the additional two components anymore
```suggestion
	{console.log('pending')}
{:then value}
	{console.log(value)}
```

Reply:
Depending on whether we intend to preserve `then` blocks, the second one should probably become just this...

```svelte
{console.log('then')}
```

...so that we can determine whether the block has been destroyed and recreated or not

Reply:
Simplifying this test doesn't really add value.

Reply:
Why? My simplification caught an issue with the previous implementation. What's the problem with it?

Reply:
I added another test, I don't think changing this test adds value as it's testing the HTML output as well as the sequence.

Reply:
I know this is very nitpicky, but could we merge both tests into one? These tests add up, and what they test is highly related, so might aswell merge them.

Reply:
The other test was updated to better reflect the different cases.